Snowflake.Data.Tests/SFConnectionIT.cs

```diff
@@ -132,13 +134,75 @@ public void TestIncorrectUserOrPasswordBasicConnection()
                 {
                     // Expected
                     logger.Debug("Failed opening connection ", e);
-                    Assert.AreEqual("08006", e.SqlState); // Connection failure
+                    Assert.AreEqual(ConnectionFailureErrorCode, e.SqlState);
                 }
 
                 Assert.AreEqual(ConnectionState.Closed, conn.State);
 			}
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestConnectionIsNotMarkedAsOpenWhenWasNotCorrectlyOpenedBefore(bool explicitClose)
+        {
+            SnowflakeDbConnectionPool.SetPooling(true);
+            SnowflakeDbConnection snowflakeConnection = null;

```

Comment:
this variable is used only inside the loop and its value doesn't preserve to the next iteration so it could be considered to declare inside the loop

Reply:
moved down to loop in this and next test